### DISCUSSION TITLE
[Question for devs] HTTP/3 and QUIC in regular mode

### DISCUSSION THREAD

**Discussion by josemariafr-upm** [mitmproxy/mitmproxy#6397] [General]
Hello everyone! First of all I'm sorry if this is in the wrong category, as the Q&A category description didn't seem right for this discussion (since I'm not looking for help as such). Secondly, let me give a bit of context:

I'm a student in university that is working on his final thesis of the Master's Degree of Telecom. Engineering. For such thesis the department is interested in intercepting and decrypting incoming and outcoming HTTP/3 (if possible) or QUIC (at worst) outside a target device but BEFORE reaching the internet (that means, then, usage of a forward proxy). 
We know (and we are interested in it) about the Regular mode in MitM Proxy, but as it is said in the documentation, [HTTP/3 is only supported in Reverse mode](https://docs.mitmproxy.org/stable/concepts-protocols/#http3), so our question for the devs (I've seen that @mhils would be one of them? Sorry for the bothering of the mention Maximilian), then: 
 
- Are you working in giving support for HTTP/3 in Regular mode? If so, do you have any estimation date of release of such support (if not but there is a private repository or something not available publicly, is there a way to at least try it?)?

Obviously we are studying alternatives to achieve that, but if there is work in progress we could wait depending on the timeline, since the interception per se is not the main objective of the thesis, so that we could save time in that aspect.

Thanks in advance to everyone in the MitMProxy team for having such a great tool! I'm open to questions if needed!

**Comment by mhils** [comment]
> Are you working in giving support for HTTP/3 in Regular mode? 

Yes, but no estimate for a release date. 

Transparent QUIC mostly works already, but when I last tested stuff I stumbled over a bunch of weird issues that made me disable it for the experimental release. You are welcome to revert https://github.com/mitmproxy/mitmproxy/commit/ff0155b1f79453c1dd444f28333fee5baf2ba3af and see how that works for you. :)

**Reply by josemariafr-upm** [reply]
Understood, we will try that commit then and see how it could be used to achieve what we want.
We'll wait to see if you are able to give an ETA someday just in case. Would it be too much to ask if you would be so kind as to let us know by replying in this thread or by creating another thread in the Discussions section?

Again thank you very much for the work!

**Reply by mhils** [reply]
I'd recommend subscribing for releases. We have an RSS feed on mitmproxy.org, or you can watch on GitHub for releases and get an email.

**Reply by josemariafr-upm** [reply]
Hi, and sorry to bother you again. I was testing the commit you mentioned to see if it fits our needs as I told you. I have changed my PC and I had to reinstall everything. The thing is, I'm having troubles with the developing environment while installing the dependencies and building everything (see attached image), and I don't know why since in the previous PC worked fine.

![image](https://github.com/mitmproxy/mitmproxy/assets/73529700/1e1a49ab-e8c7-4973-a462-09b0d4d3362f)

Just to be sure, what I'm doing to specifically execute the commit you mentioned is:
```
git clone https://github.com/mitmproxy/mitmproxy.git
cd mitmproxy
git reset --hard ff0155b
python -m venv venv
venv\Scripts\pip install -e .[dev] (here is where the problem shows)
```
Am I doing something wrong? I need to install something else? I don't know why it doesn't work this time.
Thanks in advance!

**Reply by josemariafr-upm** [reply]
I will answer myself, in case someone has the same problem.

The error is due to the Python version: I was using 3.12.0, but seems like zstandard doesn't support it yet. Uninstalling Python and reinstalling the latest 3.11 version (at this moment it's the 3.11.6) solves the issue

**Comment by josemariafr-upm** [comment]
Hi again @mhils and rest of devs. I've been trying to make work the transparent mode for QUIC packets using the commit [ff0155b](https://github.com/mitmproxy/mitmproxy/commit/ff0155b1f79453c1dd444f28333fee5baf2ba3af) that you told me but no luck, and I haven't found the reason. 
What I'm trying to do is: I'm running a Caddy server with a simple HTML index.html, and I'm doing curl requests to it (with --http3-only option to force the HTTP/3 exchange) to then capture the packets both with Wireshark and MitMProxy (at least the QUIC packets) to correlate the results and see if MitMProxy could be used for our needs in another machine. The problem is I'm not getting any QUIC packets in MitMProxy (transparent nor wireguard mode), not only requesting from the local Caddy Server but also to Google, Facebook, etc; and I know I'm supposed to see the QUIC packets there.
I have the MitMProxy CA certificate installed in both my PC and the phone I was using to test the Wireguard mode (and making the requests to the web pages from there too by connecting to the PC by creating a Wi-Fi hotspot in the phone and connecting the PC there).

So at this point, do any of you have any suggestion as to at least try to make the capture of QUIC packets work?
Thanks in advance!
